app/file.go

```diff
@@ -604,3 +604,32 @@ func (a *App) GetFileInfo(fileId string) (*model.FileInfo, *model.AppError) {
 		return result.Data.(*model.FileInfo), nil
 	}
 }
+
+func (a *App) CopyFileInfos(userId string, fileIds []string) ([]string, *model.AppError) {
+	newFileIds := []string{}
+
+	now := time.Now()
+
+	for _, fileId := range fileIds {
+		fileInfo := &model.FileInfo{}
+
+		if result := <-a.Srv.Store.FileInfo().Get(fileId); result.Err != nil {
+			return nil, result.Err
+		} else {
+			fileInfo = result.Data.(*model.FileInfo)
+		}
+
+		fileInfo.Id = model.NewId()
+		fileInfo.CreatorId = userId
+		fileInfo.CreateAt = now.UnixNano() / int64(time.Millisecond)
```

Comment:
There's a small change you can make to simplify this. If you use the `model.GetMillis()` helper function, that does the division by milliseconds for you like we use elsewhere.

You should probably also set the UpdateAt timestamp of the new file info to match the new CreateAt timestamp as well. I don't think we use it anywhere yet, but it may be used in the future

Reply:
fixed in commit d48149d